### DISCUSSION TITLE
Getting list of supported and enabled features

### DISCUSSION THREAD

**Discussion by ivenhov** [keycloak/keycloak#31148] [Q&A]
Hi

Is there a way of checking which features are enabled by default, are supported and in preview?
I'm asking because features change their status in the major releases and it quite difficult to find out what has become default over time.

As far as I can tell there are

Current status, but not for release 22, 24 etc.
https://www.keycloak.org/server/features

List of features, but without specifying 
https://www.keycloak.org/server/all-config?q=features

And at runtime, but I don't think this displays enabled by default e.g. Impersonate
/admin/master/console/#/master

I'm on Keycloak 21.1 under Docker and some of the features became enabled by default, so I would like to slim down my docker compose file or maybe disable things I don't need

Likely I'm missing something i.e. older releases docs

Thanks

**Comment by akiross** [comment]
I have the same issue: I started recently with version 24.0 since my distro had a package for it, I encountered issues because [the documentation](https://www.keycloak.org/server/all-config) is not versioned and was for v25 without signaling what options were available for each release.

When I finally migrated to v25 few days ago, everything is now a mess again because v26 is out :sweat_smile: 

I think this should be improved: it feels like keycloak is a moving target and breaking changes happen quite often, without proper documentation or a clear way on how to get all the documentation for a given release.

**Reply by shawkins** [reply]
relates to #33532

**Reply by akiross** [reply]
Right, thank you I forgot there were an issue opened
